rocksdb_admin/admin_handler.cpp

```diff
@@ -605,7 +607,7 @@ void AdminHandler::async_tm_addS3SstFilesToDB(
     return;
   }
 
-  auto responses = s3Util->getObjects(request->s3_path, local_path);
+  auto responses = s3_util_->getObjects(request->s3_path, local_path);
```

Comment:
what if someone changed s3_util_ before this line and after line 608?

I think we can do it in a simpler way. We just have a shared_ptr and a spin lock to protect it.

When we need to use it, we grab the lock, make a local copy of the shared_ptr, and release the lock. and then check if the s3 client is what we need. If yes, we just use it. Otherwise, we use it, and store it to the global shared_ptr (acquire the spin lock again)